### PR TITLE
[TASK-915] Simplify button component (even more)

### PR DESCRIPTION
**Pull request by magicznyleszek**
## Checklist

1. [ ] If you've added code that should be tested, add tests
2. [ ] If you've changed APIs, update (or create!) the documentation
3. [x] Ensure the tests pass
4. [x] Make sure that your code lints and that you've followed [our coding style](https://github.com/kobotoolbox/kpi/blob/master/CONTRIBUTING.md)
5. [x] Write a title and, if necessary, a description of your work suitable for publishing in our [release notes](https://community.kobotoolbox.org/tag/release-notes)
6. [x] Mention any related issues in this repository (as #ISSUE) and in other repositories (as kobotoolbox/other#ISSUE)
7. [ ] Open an issue in the [docs](https://github.com/kobotoolbox/docs/issues/new) if there are UI/UX changes

## Description

Minimizes the amount of button types we use in the UI. There are still some legacy buttons being used (that don't look properly), but the vast majority of buttons will now fall into 5 distinct types. This is an improvement from current 9 types, and from previous 15 types we've had for the last few years.

Improved the accessibility of Button: pending state is no longer focusable and keyboard-clickable, disabled button is no longer TAB focusable

## Notes

Things changed here:
- Changed Button component to not use `type` + `color` mix, but rather unified and simplified `type` (we have 5 types: `primary`, `secondary`, `danger`, `secondary danger`, and `text`)
- Updated **all** instances of `<Button>` to use the new type
- Updated Button stories to have a view with all the buttons types, sizes and states being displayed at the same time

## Related issues

Includes work from #5033

After merging this https://github.com/kobotoolbox/docs/issues/372 can be done

### CODE REVIEW

File: jsapp/js/components/common/button.tsx [hunk 1]
```diff
@@ -10,20 +10,18 @@ import cx from 'classnames';
...
+ * 3. danger - white text on red background
+ * 4. secondary danger - dark red text on light red background
+ * 5. text - dark blue text with no background
  */
-export type ButtonType = 'bare' | 'frame' | 'full';
-export type ButtonColor =
-  | 'blue'
-  | 'red'
-  | 'dark-blue';
+export type ButtonType = 'primary' | 'secondary' | 'danger' | 'secondary-danger' | 'text';
```
Comment (duvld) [inline jsapp/js/components/common/button.tsx:24]: Tiny nitpick. Is there a way to add a jsdoc to this type such that it shows up when auto-completing? At first glance these names are a bit vague and requires the user to read this component whereas before it was a bit more descriptive. 
image

Comment (duvld) [reply jsapp/js/components/common/button.tsx:24]: image
(Just checking it is the same on vscode also)

**Review by duvld — APPROVED** [review]
Everything else looks good to me :)

File: jsapp/js/components/common/button.tsx (hunk 1 above)
Comment (magicznyleszek) [reply jsapp/js/components/common/button.tsx:24]: There is a JSDoc comment for the line where type is defined, but I will move it to the line where button prop is - it should do the trick :ok:

**Pull request merged**